components/src/core/components/Input/TreeSelect/TreeSelect.vue

```diff
@@ -0,0 +1,545 @@
+        expandedIds.value.splice(index, 1);
+      }
+      if (option.children ? option.children.length != 0 : false) {
+        let children = option.children;
+        for (let child of children) {
+          removeOptionIdFromExpandedIdsArray(child);
+        }
+      }
+    };
+
+    const addExpandedChildrenToOptionsArray = (option: Option) => {
+      let optionIndex = optionsArr.value.findIndex(x => x.id == option.id);
+      if (optionIndex > -1) {
+        let children = optionsArr.value[optionIndex].children;
+        optionsArr.value.splice(optionIndex + 1, 0, ...children);
+      }
+    };
+
+    const removeExpandedChildrenFromOptionsArray = (option: Option) => {
+      let children = option.children ?? [];
+      for (let child of children) {
+        let index = optionsArr.value.findIndex(
+          option_ => option_.id == child.id,
+        );
+        if (index > -1) {
+          optionsArr.value.splice(index, 1);
+        }
+        if (option.children ? option.children.length != 0 : false) {
+          let childrenOfchild = option.children;
+          for (let childOfChild of childrenOfchild) {
+            removeExpandedChildrenFromOptionsArray(childOfChild);
+          }
+        }
+      }
+    };
+
+    const getIcon = (option: Option) => {
+      if (option.children ? option.children.length != 0 : false) {
+        if (expandedIds.value.includes(option.id)) {
+          return 'oxd-arrow-down';
+        } else return 'oxd-arrow-right';
+      } else return '';
+    };
+
+    const dropdownClasses = computed(() => {
+      return {
+        '--positon-bottom': props.dropdownPosition === BOTTOM,
+        '--positon-top': props.dropdownPosition === TOP,
+      };
+    });
+
+    const closeDropdownOnOutsideClick = () => {
+      closeDropdown();
+    };
+
+    const closeDropdown = () => {
+      dropdownOpen.value = false;
+    };
+
+    const ToggleSelectAll = (checkboxVal: boolean) => {
+      if (checkboxVal) {
+        isAllSelected.value = true;
+        selectAllOptions();
+      } else {
+        isAllSelected.value = false;
+        unselectAllOptions();
+      }
+    };
+
+    const selectAllOptions = () => {
+      for (let key in selectedIdsObject.value) {
+        selectedIdsObject.value[key] = true;
+      }
+    };
+
+    const unselectAllOptions = () => {
+      for (let key in selectedIdsObject.value) {
+        selectedIdsObject.value[key] = false;
+      }
+    };
+
+    const isIfAllOptionsSelected = () => {
+      for (let key in selectedIdsObject.value) {
+        if (!selectedIdsObject.value[key]) return false;
+      }
+      return true;
+    };
+
+    const selectPreSelectedOptions = () => {
+      for (let selectedId of props.preSelectedIds) {
+        if (selectedIdsObject.value[selectedId] != undefined) {
+          let option = findOptionByOptionId(selectedId, optionsArr.value);
+          if (typeof option !== 'string') {
+            addChildrenToSelectedIdsArray(option);
```

Comment:
This logic is similar to the logic where we select a check box. So better reuse the same.